addons/mail/static/src/core/web/mail_composer_template_selector.js

```diff
@@ -0,0 +1,139 @@
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+import { sprintf } from "@web/core/utils/strings";
+import { standardFieldProps } from "@web/views/fields/standard_field_props";
+import { useService } from "@web/core/utils/hooks";
+
+import { Component, useState, onWillStart } from "@odoo/owl";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { SelectCreateDialog } from "@web/views/view_dialogs/select_create_dialog";
+
+
+export class MailComposerTemplateSelector extends Component {
+    static template = "mail.MailComposerTemplateSelector";
+    static components = { Dropdown, DropdownItem };
+    static props = { ...standardFieldProps };
+
+    setup() {
+        this.action = useService("action");
+        this.orm = useService("orm");
+        this.state = useState({});
+        this.limit = 5;
```

Comment:
Why 5 ?

Reply:
`/roll 1d10` 🧙
GM: "5... Well... Hmmm... Your 12 points of charisma are not enough to convince your reviewer to give you a r+ right away. Your reviewer likes your PR but he's still skeptical about some changes you made."

More seriously: The dropdown menu is small and should not list much items. 5 seems like a good limit. Designers seems to recommend having list of "seven plus or minus two" items.

Reply:
Well it works only in demo. In real production environment, 5 is useless. We need more records and search capability to find the right one. Just try to find a template in odoo.com and you will understand the issue :D .